### PR TITLE
feat(passwords and secrets mechanism): a88baa34-e2ad-44ea-ad6f-8cac87bc7c71 trigger all the rules defined inside the regex_rules.json

### PR DESCRIPTION
**Pull request by rafaela-soares**
**Proposed Changes**
- **Exclude Passwords and Secrets query (a88baa34-e2ad-44ea-ad6f-8cac87bc7c71) through --exclude-queries**: this approach took advantage of disableSecretsQuery verification. It was added verification to not include the Password and Secrets mechanism when the --exclude-queries points to Passwords and Secrets query (a88baa34-e2ad-44ea-ad6f-8cac87bc7c71) and --secrets-regexes-path is not used

- **Include Passwords and Secrets query (a88baa34-e2ad-44ea-ad6f-8cac87bc7c71) through --include-queries**: if --include-queries points to the Passwords and Secrets query (a88baa34-e2ad-44ea-ad6f-8cac87bc7c71), it should trigger all the rules defined inside the regex_rules.json. For that, we have some cases:
  - --include-queries points to the Passwords and Secrets query (a88baa34-e2ad-44ea-ad6f-8cac87bc7c71) and --secrets-regexes-path is used. This is, the user wants to merge all KICS regexes with the ones pointed by --secrets-regexes-path
  - --include-queries points to the Passwords and Secrets query (a88baa34-e2ad-44ea-ad6f-8cac87bc7c71) and --secrets-regexes-path is not used. This is, the user only wants to include the KICS regexes

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: pkg/engine/secrets/inspector.go [hunk 1]
```diff
@@ -95,8 +95,10 @@ func NewInspector(
 	disableSecretsQuery bool,
 	executionTimeout int,
 	regexRulesContent string,
+	isCustomSecretsRegexes bool,
 ) (*Inspector, error) {
-	if disableSecretsQuery {
+	excludeSecretsQuery := isValueInArray("a88baa34-e2ad-44ea-ad6f-8cac87bc7c71", queryFilter.ExcludeQueries.ByIDs)
```
Comment (rogeriopeixotocx) [inline pkg/engine/secrets/inspector.go:100]: I'd suggest getting the query ID from here https://github.com/Checkmarx/kics/blob/master/assets/assets.go#L12

File: pkg/engine/secrets/inspector.go [hunk 2]
```diff
@@ -177,12 +179,28 @@ func (c *Inspector) Inspect(ctx context.Context, basePaths []string,
...
 }
 
-func compileRegexQueries(queryFilter *source.QueryInspectorParameters, allRegexQueries []RegexQuery) ([]RegexQuery, error) {
+func compileRegexQueries(queryFilter *source.QueryInspectorParameters, allRegexQueries []RegexQuery, isCustom bool) ([]RegexQuery, error) {
 	var regexQueries []RegexQuery
+	var includeSpecificSecretQuery bool
+
+	allSecretsQueryAndCustom := false
+
+	includeAllSecretsQuery := isValueInArray("a88baa34-e2ad-44ea-ad6f-8cac87bc7c71", queryFilter.IncludeQueries.ByIDs)
```
Comment (rogeriopeixotocx) [inline pkg/engine/secrets/inspector.go:188]: I'd suggest getting the query ID from here https://github.com/Checkmarx/kics/blob/master/assets/assets.go#L12

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]

**Comment by rogeriopeixotocx** [comment]
add information about this here https://docs.kics.io/latest/secrets/#flags

File: pkg/engine/secrets/inspector.go (hunk 1 above)
Comment (rafaela-soares) [reply pkg/engine/secrets/inspector.go]: Great suggestion, @rogeriopeixotocx! 🚀

**Review by rogeriopeixotocx — APPROVED** [review]
LGTM

**Review by nunoocx — APPROVED** [review]
LGTM

**Pull request merged**